### PR TITLE
adding gmm-feedstock

### LINKED ISSUE #15576: [Package request] GMM++
# Some links to get started
- [Currently available conda-forge packages](https://conda-forge.org/feedstock-outputs/)
- [The conda-forge docs](https://conda-forge.org/docs/maintainer/adding_pkgs.html#the-staging-process)
- [Grayskull](https://github.com/conda-incubator/grayskull) - to automatically generate a recipe for packages on PyPI
- [The conda r skeleton helpers](https://github.com/bgruening/conda_r_skeleton_helper) - to automatically generate a recipe for packages on CRAN

# The requested package

- _Package name_: gmm++
- _Version (If a specific one is required)_: 5.0, and newer 
- _URL_: http://getfem.org/gmm/index.html, http://getfem.org/download.html, http://download-mirror.savannah.gnu.org/releases/getfem/stable/
- _Available via (GitHub Releases, PyPI, etc.)_: http://download-mirror.savannah.gnu.org/releases/getfem/stable/
- _Additional comments_: unrelated to r-gmm. Aka getFEM, but only the c++ library is needed for now but there is a python interface/bindings that could be added later. I am not that experienced with cmake/c++ building but the http://getfem.org/gmm/install.html guide makes it look very easy to compile so maybe it would be easy to add so if that is the case let me know and I can maybe add it if I get some guidance. 

# Before submitting
- [x] Verify that the package does not already exist (Also check with _hyphens_ vs. _underscores_ and added identifiers like `python-` and `r-`).
- [x] Check that no previous [issue](https://github.com/conda-forge/staged-recipes/issues) exists and that no [PR](https://github.com/conda-forge/staged-recipes/pulls) is currently open.

### PR DESCRIPTION
**Pull request by AndrewAnnex**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

fixes #15576

### CODE REVIEW

**Comment by AndrewAnnex** [comment]
@conda-forge/help-c-cpp ready for review

File: recipes/gmm/build.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+# from https://web.archive.org/web/20200225211624/http://getfem.org/gmm/install.html
+# configure first using conda prefix
+./configure --prefix=${PREFIX}
+# make it
+make
```
Comment (xhochy) [inline recipes/gmm/build.sh:7]:
```suggestion
make -j$CPU_COUNT
```

File: recipes/gmm/build.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+# from https://web.archive.org/web/20200225211624/http://getfem.org/gmm/install.html
+# configure first using conda prefix
+./configure --prefix=${PREFIX}
+# make it
```
Comment (xhochy) [inline recipes/gmm/build.sh:6]: These comments don't seem helpful 🤣

Comment (AndrewAnnex) [reply recipes/gmm/build.sh:6]: yeah this one isn't I suppose, but as a person who doesn't use the c/c++ stack of cmake/autotools/gcc/make day-to-day some of the earlier ones are more helpful for me personally. Conda-forge could use more docs for contributing c/c++ packages in general (or at least link to conda-build more). Obvious to some but not to all.

File: recipes/gmm/LICENSE.txt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+Getfem++  is  free software;  you  can  redistribute  it  and/or modify it
```
Comment (isuruf) [inline recipes/gmm/LICENSE.txt:1]: Why is this file needed here? Isn't it in the source tarball?

File: recipes/gmm/build.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+make -j$CPU_COUNT
```
Comment (isuruf) [inline recipes/gmm/build.sh:7]: Is there a `make check` we can run here?

File: recipes/gmm/LICENSE.txt (hunk 3 above)
Comment (AndrewAnnex) [reply recipes/gmm/LICENSE.txt:1]: there is a "COPYING" file, if I can use that instead let me know

File: recipes/gmm/build.sh (hunk 4 above)
Comment (AndrewAnnex) [reply recipes/gmm/build.sh:7]: looks like there is a `make check` and also a `make installcheck`, I didn't know if there was a check target to use, anyways I'll add the `make check`

File: recipes/gmm/LICENSE.txt (hunk 3 above)
Comment (AndrewAnnex) [reply recipes/gmm/LICENSE.txt:1]: I think I can per the conda forge docs, so I'll try that now and if the linters complain I'll undo it

**Review by xhochy — APPROVED** [review]
